sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -112,4 +112,20 @@ class Utils {
     }
   }
 
+  /**
+   * Called when an API call fails, so that a connection error message can be shown.
+   */
+  static connectionFailed() {
```

Comment:
Here's an alternative approach: ApiManager now has handlers for connected and disconnected that can be set by its caller. This way it's separated from its usage, and the containing element, `datalab-app` in this case, can listen for changes in connected status.

A better way to do this would be to observe the value change on the `isConnected` property, but I didn't want to add the complication of observer code, so I'm allowing exactly one listener until we have the need for multiple.